chainstate/types/src/block_status.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use enum_iterator::Sequence;
+use serialization::{Decode, Encode};
+
+/// Block validation steps are always performed in the same order, which is represented by this enum.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Sequence, Decode, Encode)]
+pub enum BlockValidationStage {
+    Initial,
```

Comment:
I'm somewhat hesitant over this but I have nothing better to offer

Reply:
`NotValidated`/`UnValidated`/`NotChecked`? I don't know some seem too close to `Invalid` which may be confusing. Agreed this should be something more explicit about the fact that the block has not been validated, `Initial` does not quite cut it.

Reply:
I've changed it to Unchecked, because "checked" is already used in FullyChecked (and IMO Unchecked sounds better than NotChecked). Let me know if this is OK.

Reply:
> I've changed it to Unchecked, because "checked" is already used in FullyChecked (and IMO Unchecked sounds better than NotChecked). Let me know if this is OK.

Sounds good to me